server/data/supabase_server_client.ts

```diff
@@ -0,0 +1,25 @@
+"use server";
+
+import { cookies } from "next/headers";
+import { createServerClient } from "@supabase/ssr";
+import { SupabaseClient } from "@supabase/supabase-js";
+import { Database } from "@/data/supabaseTypes";
+
+export async function buildSupabaseServerClient(): Promise<
```

Comment:
+1 for isolating client construction. What is the difference between `createClient` and `createServerClient` API?

Reply:
the `"use server";` line at the top of the file assures that this is handled server side (and not client side) which is useful for other functions that we require to be server side (for example, validating keys that shouldn't be exposed to the client.)

client here is a bit overloaded. there is both the `supabaseClient` that is a class that acts essentially as a ORM, and the browser, used in the `"use client";` / `"use server";` distinction.

Reply:
oh and also, the `supabaseClient` is the more general type, which `supabaseServerClient` returns (but is only used server side)